### PR TITLE
Use a different SSH key per site

### LINKED ISSUE #179: Use an SSH key per site
So that builds can only access their own site.

### PR DESCRIPTION
**Pull request by singpolyma**
Needs to generate a key per site and then actually feed that key to git
during operations.

Closes #179

### CODE REVIEW

**Comment by ericroberts** [comment]
Has conflicts

File: app/site.rb [hunk 1]
```diff
@@ -81,6 +99,7 @@ def resource(path)
 		end
 
 		def git
+			ENV["SITE_PUBLIC_KEY_PATH"] = ssh_private_key_path.to_s
```
Comment (ericroberts) [inline app/site.rb]: Should this be <code>ENV["SITE_PUBLIC_KEY_PATH"] = ssh_*public*_key_path.to_s</code>?

Comment (singpolyma) [inline app/site.rb]: The env var is actually the one named wrong.  I've updated it

**Comment by singpolyma** [comment]
@ericroberts fixed conflicts

**Pull request merged**